src/devices/esp8266.c

```diff
@@ -123,6 +123,32 @@ static void ipd_urc_cb(char *msg)
 		state.hooks.data_received_cb(chan_id, len, data);
 }
 
+static void wifi_action_callback(const char* msg)
+{
+	/* Check for the hook */
+	if (!state.hooks.client_state_changed_cb)
+		return;
+
+	enum client_action action;
+	/* Look at our 5th character to determin our action */
+	switch (msg[5]) {
+	case 'C':
```

Comment:
Nit suggestion: would be cool to include, above each switch statement a comment with the actual message from the ESP8266 that we're capturing. for reference.